### PR TITLE
fixes some bugs

### LINKED ISSUE #19305: Nuclear operative leaders don't have huds
### Severity

Inconvenience

### Short Description

Nuclear operative leader doesn't have an icon next to them which leads to nukies killing each other sometimes

### Reproduction

1. Nukies roll for the round start
2. Become a leader of the team
3. You don't have an icon next to you
4. possibly get killed by other Nuke ops because they didn't notice that you don't have a hud before going to the station
5. Death

### Round ID

Round ID: 47702

### Test Merges

#18498 #19287 #18502

### BYOND Username

Kmla

### Was this done on a locally hosted or non-Yogstation hosted server?

No

### Additional information

_No response_

### LINKED ISSUE #19319: Anomalyus Crystal lies
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: The possession effect of the crystal lies; It claims that you can leave the mob at will, and you die if it dies. what instead happens is you just fucking die when you use the ablity to leave the mob it just gibs you and deletes your body

 Submitted by: poledancingcatgirl (605167460450304019)

### LINKED ISSUE #19374: space dragon has TOO MUCH STOMACH ACID
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: when space dragon is butchered after dying the bodies inside are just deleted instead of dropping (they get deleted precisely when its butchered if thats any help)

 Submitted by: witherc6 (429012660869136394)

### LINKED ISSUE #19394: all seeing predator doesnt fuckign work
## Test Merges: 
Not Supplied / None
## Reproduction:
i think the all seeing predator button for holoparas predator ability is meant to do something

 Submitted by: snoobper (1079798074068709386)

### LINKED ISSUE #19402: Xeno spit can only shoot up
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: You aim at the side, it shoots up, you aim below you, it shoots up

 Submitted by: pibecantina (341102461504716800)

### LINKED ISSUE #19436: TC GOLEM IS BROKE
## Test Merges: 
Not Supplied / None
## Reproduction:
Description:TC golem ability does not work, i can cast it and it will do the particle effect the teleportation has, but it wont actually teleport and it will go in cooldown regardless

 Submitted by: pibecantina (341102461504716800)

### PR DESCRIPTION
**Pull request by tattax**
closes: https://github.com/yogstation13/Yogstation/issues/19436
closes: https://github.com/yogstation13/Yogstation/issues/19402
closes: https://github.com/yogstation13/Yogstation/issues/19394
closes: https://github.com/yogstation13/Yogstation/issues/19374
closes: https://github.com/yogstation13/Yogstation/issues/19319
closes: https://github.com/yogstation13/Yogstation/issues/19305

# Changelog
:cl:  
bugfix: fixed a few abilities that dind't work (holos, golem, queen, a. crystal)
bugfix: fixed hud issue with nuke ops
/:cl:

### CODE REVIEW

**Comment by neeshacark** [comment]
can you fix drones not being able to put stuff in their bags with E anymore (i *think* you caused this)

File: code/datums/elements/content_barfer.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Content Barfer; which expels the contents of a mob when it dies, or is transformed
+ *
+ * Used for morphs and bileworms!
```
Comment (JohnFulpWillard) [inline code/datums/elements/content_barfer.dm]:
```suggestion
 * Used for morphs!
```

File: code/datums/elements/content_barfer.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignals(target, list(COMSIG_LIVING_DEATH, COMSIG_LIVING_ON_WABBAJACKED), PROC_REF(barf_contents))
+
+/datum/element/content_barfer/Detach(datum/target)
+	UnregisterSignal(target, list(COMSIG_LIVING_DEATH, COMSIG_LIVING_ON_WABBAJACKED))
+	return ..()
+
+/datum/element/content_barfer/proc/barf_contents(mob/living/target)
+//	SIGNAL_HANDLER
```
Comment (JohnFulpWillard) [inline code/datums/elements/content_barfer.dm:22]:
```suggestion
	SIGNAL_HANDLER
```

Comment (tattax) [reply code/datums/elements/content_barfer.dm:22]: doesn't work, rather leave it like this then fix forcemove sometime later

**Review by JohnFulpWillard — APPROVED** [review]

**Comment by milk-is-cool** [comment]
> can you fix drones not being able to put stuff in their bags with E anymore (i _think_ you caused this)

nah this has been an issue since before the antag pr cause its always been a problem any time im a cogs creb

**Pull request merged**